### PR TITLE
[PyMySQL] Add __enter__/__exit__ to pymysql.Connection

### PR DESCRIPTION
**Pull request by andrewsoutar**
See #7044

### CODE REVIEW

File: stubs/PyMySQL/pymysql/connections.pyi [hunk 1]
```diff
@@ -195,6 +196,8 @@ class Connection(Generic[_C]):
     def get_proto_info(self): ...
     def get_server_info(self): ...
     def show_warnings(self): ...
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(self, *exc_info: Any) -> None: ...
```
Comment (srittau) [inline stubs/PyMySQL/pymysql/connections.pyi:200]: As the contents of `exc_info` are ignored:

```suggestion
    def __exit__(self, *exc_info: object) -> None: ...
```

**Review by srittau — CHANGES_REQUESTED** [review]

File: stubs/PyMySQL/pymysql/connections.pyi (hunk 1 above)
Comment (JelleZijlstra) [inline stubs/PyMySQL/pymysql/connections.pyi:200]:
```suggestion
    def __exit__(self, *exc_info: object) -> None: ...
```
Prefer object over Any when the function accepts all objects.

Comment (srittau) [reply stubs/PyMySQL/pymysql/connections.pyi]: Slowpoke 😁

**Review by srittau — APPROVED** [review]

File: stubs/PyMySQL/pymysql/connections.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stubs/PyMySQL/pymysql/connections.pyi]: I failed today

**Pull request merged**